matter-labs/zksync-era
# feat(basic_witness_input_producer): Witness inputs queued after BWIP run

# What ❔

Add supporting transition state for `witness_input` jobs. Prior to this PR, `MetadataCalculator` would create `witness_input` jobs that are ready to be processed by `WitnessGenerator`s. After this PR, `WitnessGenerator`s will need to wait for `BasicWitnessInputProducer` to generate it's artifacts as well. This PR must be deployed after BWIP is deployed in the given environment.

## Why ❔

Necessary for transitioning to Witness Generation from Blob.

## Other remarks

Deployment will be rather involved. @EmilLuta will own it with support from @Deniallugo, starting 31st of Oct, 2023. Steps:
```
for env in ['stage2', 'testnet', 'boojnet', 'mainnet2'] {
  - Deploy to <ENV> BWIP.
  - Wait for all `witness_input` jobs to be finished prior to BWIP's processing (or mark them as manually skipped)
  - Deploy to <ENV> this patch.
  - Turn on Shadow Proving
  - Observe
}
```

## Checklist
